tegra/tegra.c

```diff
@@ -206,7 +206,7 @@ int drm_tegra_bo_unref(struct drm_tegra_bo *bo)
 		return -EINVAL;
 
 	if (!atomic_dec_and_test(&bo->ref))
-		return 0;
+		return atomic_read(&bo->ref);
```

Comment:
I'm still unsure about it. The performance aspect is questionable since assert() contribution is miserable. What about printing a error message to stderr?